File: packages/genai-speech-recognition/ios/Plugin/GenAiSpeechRecognitionPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(GenAiSpeechRecognitionPlugin, "GenAiSpeechRecognition",
           CAP_PLUGIN_METHOD(checkFeatureStatus, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(checkPermissions, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(downloadFeature, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(requestPermissions, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(startRecognition, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(stopRecognition, CAPPluginReturnPromise);
)

File: packages/genai-speech-recognition/ios/Plugin/GenAiSpeechRecognitionPlugin.swift
import Foundation
import Capacitor

@objc(GenAiSpeechRecognitionPlugin)
public class GenAiSpeechRecognitionPlugin: CAPPlugin {
    @objc func checkFeatureStatus(_ call: CAPPluginCall) {
        rejectCallAsUnavailable(call)
    }

    @objc override public func checkPermissions(_ call: CAPPluginCall) {
        rejectCallAsUnavailable(call)
    }

    @objc func downloadFeature(_ call: CAPPluginCall) {
        rejectCallAsUnavailable(call)
    }

    @objc override public func requestPermissions(_ call: CAPPluginCall) {
        rejectCallAsUnavailable(call)
    }

    @objc func startRecognition(_ call: CAPPluginCall) {
        rejectCallAsUnavailable(call)
    }

    @objc func stopRecognition(_ call: CAPPluginCall) {
        rejectCallAsUnavailable(call)
    }

    private func rejectCallAsUnavailable(_ call: CAPPluginCall) {
        call.unavailable("This method is not available on this platform.")
    }
}
